rust/nix-store/src/path.rs

```diff
@@ -0,0 +1,33 @@
+use anyhow::Result;
+use nix_c_raw as raw;
+use nix_util::string_return::{callback_get_vec_u8, callback_get_vec_u8_data};
+
+pub struct StorePath {
+    raw: *mut raw::StorePath,
+}
+impl StorePath {
+    pub fn new_raw_clone(raw: *const raw::StorePath) -> Self {
+        Self::new_raw(unsafe { raw::store_path_clone(raw as *mut raw::StorePath) })
```

Comment:
Does this mean when we use `new_raw_clone()` we're still expected to free the original, or is the whole point of this function that sometimes we get a `raw::StorePath` that we don't own?

Reply:
Both are true, but the latter is the reason to have this. The store paths from the realised string are borrowed. I'm adding docs for this.